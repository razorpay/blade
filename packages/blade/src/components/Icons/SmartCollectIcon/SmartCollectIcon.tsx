import { Svg, Path } from '../_Svg';
import type { IconComponent } from '..';
import useIconProps from '../useIconProps';

const SmartCollectIcon: IconComponent = ({ size, color, ...styledProps }) => {
  const { height, width, iconColor } = useIconProps({ size, color });

  return (
    <Svg {...styledProps} width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M22 9C22.5523 9 23 8.55228 23 8V4C23 2.34315 21.6569 1 20 1H16C15.4477 1 15 1.44772 15 2C15 2.55228 15.4477 3 16 3H20C20.5523 3 21 3.44772 21 4V8C21 8.55228 21.4477 9 22 9ZM1 16C1 15.4477 1.44772 15 2 15C2.55228 15 3 15.4477 3 16V20C3 20.5523 3.44772 21 4 21H8C8.55228 21 9 21.4477 9 22C9 22.5523 8.55228 23 8 23H4C2.34315 23 1 21.6569 1 20V16ZM23 16C23 15.4477 22.5523 15 22 15C21.4477 15 21 15.4477 21 16V20C21 20.5523 20.5523 21 20 21H16C15.4477 21 15 21.4477 15 22C15 22.5523 15.4477 23 16 23H20C21.6569 23 23 21.6569 23 20V16ZM1 8C1 8.55228 1.44772 9 2 9C2.55228 9 3 8.55228 3 8V4C3 3.44772 3.44772 3 4 3H8C8.55228 3 9 2.55228 9 2C9 1.44772 8.55228 1 8 1H4C2.34315 1 1 2.34315 1 4V8ZM16 7H8V11H16V7ZM18 11V7C18 5.89543 17.1046 5 16 5H8C6.89543 5 6 5.89543 6 7V11H5C4.44772 11 4 11.4477 4 12C4 12.5523 4.44772 13 5 13H6V17C6 18.1046 6.89543 19 8 19H16C17.1046 19 18 18.1046 18 17V13H19C19.5523 13 20 12.5523 20 12C20 11.4477 19.5523 11 19 11H18ZM16 13H8V17H16V13Z"
        fill={iconColor}
      />
    </Svg>
  );
};

export default SmartCollectIcon;
