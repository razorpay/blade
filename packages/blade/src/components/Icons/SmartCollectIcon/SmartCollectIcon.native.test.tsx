import SmartCollectIcon from '.';
import renderWithTheme from '~utils/testing/renderWithTheme.native';

describe('<SmartCollectIcon />', () => {
  it('should render SmartCollectIcon', () => {
    const renderTree = renderWithTheme(
      <SmartCollectIcon color="feedback.icon.neutral.intense" size="large" />,
    ).toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
