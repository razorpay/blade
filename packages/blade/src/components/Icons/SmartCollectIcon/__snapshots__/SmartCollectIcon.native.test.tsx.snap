// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmartCollectIcon /> should render SmartCollectIcon 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSVGSvgView
    accessibilityElementsHidden={true}
    align="xMidYMid"
    bbHeight="20px"
    bbWidth="20px"
    data-blade-component="icon"
    fill="none"
    focusable={false}
    height="20px"
    importantForAccessibility="no-hide-descendants"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {},
        ],
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="20px"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M14.7907 6.7619C14.7907 8.33986 13.5413 9.61905 12 9.61905C10.4587 9.61905 9.2093 8.33986 9.2093 6.7619C9.2093 5.18395 10.4587 3.90476 12 3.90476C13.5413 3.90476 14.7907 5.18395 14.7907 6.7619ZM16.6512 6.7619C16.6512 9.39183 14.5688 11.5238 12 11.5238C9.43123 11.5238 7.34884 9.39183 7.34884 6.7619C7.34884 4.13198 9.43123 2 12 2C14.5688 2 16.6512 4.13198 16.6512 6.7619ZM2 9.61905C2 8.92858 2.6617 8.66667 3.16279 8.66667C3.66388 8.66667 4.32558 8.9881 4.32558 9.67857V15.0952L6.90552 17.7292C7.15183 17.981 7.53233 17.9597 7.77762 17.7068L8.27907 17.1935C8.52435 16.9406 8.52622 16.5367 8.27907 16.2857L6.09157 14.0536C5.82397 13.7796 5.826 13.34 6.09157 13.064L6.63663 12.4985C6.9022 12.2224 7.3138 12.2022 7.5814 12.4762L10.3721 15.3333V21C10.3721 21.5523 9.92438 22 9.37209 22H6.02326C5.47097 22 5.02326 21.5523 5.02326 21V19.381L2 16.0476V9.61905ZM19.6744 9.67857C19.6744 8.9881 20.3361 8.66667 20.8372 8.66667C21.3383 8.66667 22 8.92858 22 9.61905V16.0476L18.9767 19.381V21C18.9767 21.5523 18.529 22 17.9767 22H14.6279C14.0756 22 13.6279 21.5523 13.6279 21V15.3333L16.4186 12.4762C16.6862 12.2022 17.0978 12.2224 17.3634 12.4985L17.9084 13.064C18.174 13.34 18.176 13.7796 17.9084 14.0536L15.7209 16.2857C15.4738 16.5367 15.4756 16.9406 15.7209 17.1935L16.2224 17.7068C16.4677 17.9597 16.8482 17.981 17.0945 17.7292L19.6744 15.0952V9.67857Z"
        fill={
          {
            "payload": 4280563015,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
